tasks/Module 3/task_2_test.py

```diff
@@ -44,5 +41,11 @@ def test_two_chars_and_separator():
 
 def test_without_separator():
     """Test function without separator provided."""
+    assert my_splitter("string with !@#$double spaces") == \
```

Comment:
Why does "string with !@#$double spaces" have singe spaces? :)

I think there should be two case for string with double spaces:

- test if `my_splitter("string  with  !@#$double  spaces and new\nline")` returns `["string", "with", "!@#$double", "spaces", "and", "new", "line"]` - empty strings are filtered out
- test if `my_splitter("string  with  !@#$double  spaces", separator=" ")` returns `["string", "", "with", "", "!@#$double", "", "spaces"]`

Reply:
Please also add a test case then string ends with separator:
`my_splitter("aaa,bbb,", ",")`
It is interesting case that shows small bug in your algorithm :)

Reply:
@musk-ox infinite loop is not a small bug ))
as for the tests, i thought that we need to have two cases;
1) when in first we are not providing a separator and words separated by one space
2) the safe as above, but words separated with two spaces

I've fixed this alread, with below:

`def test_without_separator():`
   ` """Test function without separator provided."""`
    `assert my_splitter("string  with  !@#$double  spaces") == \`
           `["string", "with", "!@#$double", "spaces"]`


`def test_double_spaces():`
    `"""Test function with with string with double spaces and space`
   ` as a separator"""`
    `assert my_splitter("string  with  !@#$double  spaces", " ") == \`
           `["string", "", "with", "", "!@#$double", "", "spaces"]`

Reply:
> infinite loop is not a small bug

I did't want to cause panic :)

> I've fixed this alread, with below 

Did you push it? I cannot see it here.

Reply:
@musk-ox no, i still looking for a solution with `my_splitter("aaa,bbb,", ",")` xD

Reply:
new implementation introduced based on above